ui/src/app/views/concept-table/component.tsx

```diff
@@ -0,0 +1,140 @@
+    const selectedVocabularies =
+        toggleIncludes(vocabulary, this.state.selectedVocabularies) as unknown as string[];
+    this.filterImageSrc = selectedVocabularies.length > 0 ? 'filtered' : 'filter';
+    this.dt.filter(selectedVocabularies, 'vocabularyId', 'in');
+    this.setState({selectedVocabularies: selectedVocabularies});
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // The purpose of this is to reset the filter on vocabulary on change of domain/concepts
+    if (nextProps.concepts !==  this.props.concepts) {
+      if (this.state.selectedVocabularies) {
+        this.dt.filter([], 'vocabularyId', 'in');
+        this.setState({selectedVocabularies : []});
+      }
+    }
+  }
+
+  render() {
+    const vocabularyFilter = <PopupTrigger
+        side='bottom'
+        content={
+          this.distinctVocabulary().map((vocabulary, i) => {
+            return <div key={i}>
+              <CheckBox style={{marginLeft: '0.2rem', marginRight: '0.3rem'}}
+                        checked={fp.includes(vocabulary, this.state.selectedVocabularies)}
+                        onChange={(checked) => this.filterByVocabulary(vocabulary)}>
+              </CheckBox>
+              <label style={{marginRight: '0.2rem'}}>{vocabulary}</label></div>;
+          })}>
+      <Clickable
+          data-test-id='workspace-menu-button'
+          hover={{opacity: 1}}>
+        <img style={{width: '15%', marginLeft: '-2.5rem'}}
+             src={'/assets/icons/' + this.filterImageSrc + '.svg'}/>
+      </Clickable>
+    </PopupTrigger>;
+    return <div data-test-id='conceptTable'>
+      <DataTable ref={(el) => this.dt = el} value={this.props.concepts}
+                 paginator={true} rows={50} scrollable={true} loading={this.props.loading}
+                 selection={this.state.selectedConcepts}
+                 onSelectionChange={e => this.updateSelectedConceptList(e.value)} >
+      <Column bodyStyle={{...styles.colStyle, width: '3rem'}} headerStyle = {{width: '3rem'}}
+              selectionMode='multiple' />
+      <Column bodyStyle={styles.colStyle} field='conceptName' header='Name'/>
+      <Column bodyStyle={styles.colStyle} field='conceptSynonyms' header='Synonyms'
+              body={this.conceptSynonymColTemplate}/>
+      <Column bodyStyle={styles.colStyle} field='conceptCode' header='Code'/>
+      <Column field='vocabularyId' header='Vocabulary' bodyStyle={styles.colStyle}
+              filter={true} headerStyle={{display: 'flex', textAlign: 'center', paddingTop: '0.6rem'
+        , paddingLeft: '5rem'}}
+              filterElement={vocabularyFilter} />
+      <Column style={styles.colStyle} field='countValue' header='Count'/>
+    </DataTable>
+    </div>;
+  }
+}
+
+@Component({
+  selector: 'app-concept-table',
+  template: '<div #root></div>'
+})
+export class ConceptTableComponent extends ReactWrapperBase {
+  @Input() concepts: Object[];
+  @Output() getSelectedConcepts = new EventEmitter<any>(true);
```

Comment:
generally in these conversions we haven't been using `@Output`, but rather putting a callback function in the parent component and passing it in as an input prop.   i think we should keep using that convention here. we can talk faces if this is confusing or if there's a specific reason why we don't want to do that in this case